### PR TITLE
Re:Connect - enabling building reconnect uri

### PR DESCRIPTION
**Pull request by moozzyk**

### CODE REVIEW

File: src/signalrclient/url_builder.cpp [hunk 1]
```diff
@@ -81,19 +102,34 @@ namespace signalr
...
         }
 
+        web::uri build_reconnect(const web::uri& base_url, transport_type transport, const utility::string_t& connection_token,
+            const utility::string_t& connection_data, const utility::string_t& last_message_id, const utility::string_t& groups_token,
+            const utility::string_t& query_string)
+        {
+            _ASSERTE(connection_token.length() > 0);
+
+            auto builder = build_uri(base_url, _XPLATSTR("reconnect"), transport, connection_token, connection_data, query_string, last_message_id, groups_token);
+            return convert_to_websocket_url(builder, transport).to_uri();
```
Comment (BrennanConroy) [inline src/signalrclient/url_builder.cpp]: This will have to change when we support other transport types?

Comment (BrennanConroy) [inline src/signalrclient/url_builder.cpp]: Nevermind, it takes care of that internally.

File: test/signalrclienttests/url_builder_tests.cpp [hunk 2]
```diff
@@ -101,6 +101,118 @@ TEST(url_builder_connect, url_correct_if_connection_data_not_empty)
...
+
+    ASSERT_EQ(
+        web::uri(_XPLATSTR("ws://fake/signalr/reconnect?transport=webSockets&clientProtocol=1.4&connectionToken=connection%20token&connectionData=%5B%7B%22Name%22:%22ChatHub%22%7D%5D&groupsToken=G%207")),
+        url_builder::build_reconnect(web::uri{ _XPLATSTR("http://fake/signalr/") }, transport_type::websockets,
+        _XPLATSTR("connection token"), _XPLATSTR("[{\"Name\":\"ChatHub\"}]"), _XPLATSTR(""), _XPLATSTR("G 7"), _XPLATSTR("")));
+}
+
+TEST(url_builder_reconnect, query_string_added_after_message_id_and_groups_token)
+{
+    auto x = url_builder::build_reconnect(web::uri{ _XPLATSTR("http://fake/signalr/") }, transport_type::long_polling,
```
Comment (BrennanConroy) [inline test/signalrclienttests/url_builder_tests.cpp]: Unused variable

**Comment by BrennanConroy** [comment]
Looks good :ship: :it:

File: test/signalrclienttests/url_builder_tests.cpp (hunk 2 above)
Comment (moozzyk) [inline test/signalrclienttests/url_builder_tests.cpp]: The whole line is not needed. I added it when debugging (could not get the expression to evaluate in the watch window) and forgot to remove. Will remove.

**Comment by moozzyk** [comment]
Thanks @BrennanConroy !

**Pull request merged**